docs/html-templates.md

```diff
@@ -0,0 +1,40 @@
+# Html templates
+
+### Behavior
+
+* Templates are automatically recompiled in development when the file
+is saved; a refresh of the page should be enough to display the
+latest version.
+
+* The context, which sets the values that the variables in HTML
+templates should take, is defined in `zerver/context_processors.py`.
+Additional variable-setting might take place in the render call for
+that specific template.
```

Comment:
This part about `context_processors.py` is only true for backend templates.  So I think we should move a version of this paragraph into the below sections.

Would also clarify this a bit more by giving examples in each case of the places one might find context.  E.g. for backend, it's: 

* context_processors.py 
* The arguments to the `render` call in the relevant view function that renders the template.  Easy to find the code that does this by e.g. `git grep zerver/apps.html`, ....  
* (Context can be specified in `zproject/urls.py` for some fairly static pages where we're using Django's TemplateView)

Would also try to write this in a more newbie-friendly fashion, e.g. a better written version of "You can use variables in templates.  The variables available to the template are called the **context**.  ..."